File: pdu/python/sensor_msgs/pdu_cdr_conv_MultiEchoLaserScan.py
from .pdu_pytype_MultiEchoLaserScan import MultiEchoLaserScan
from ..pdu_cdr_runtime import CdrReader, CdrWriter

from ..std_msgs.pdu_cdr_conv_Header import *
from ..sensor_msgs.pdu_cdr_conv_LaserEcho import *
from ..builtin_interfaces.pdu_cdr_conv_Time import *



def py_to_cdr_body_MultiEchoLaserScan(writer: CdrWriter, src: MultiEchoLaserScan):
    py_to_cdr_body_Header(writer, src.header)
    writer.write_float32(src.angle_min)
    writer.write_float32(src.angle_max)
    writer.write_float32(src.angle_increment)
    writer.write_float32(src.time_increment)
    writer.write_float32(src.scan_time)
    writer.write_float32(src.range_min)
    writer.write_float32(src.range_max)
    writer.write_sequence_length(src.ranges)
    for elem in src.ranges:
        py_to_cdr_body_LaserEcho(writer, elem)
    writer.write_sequence_length(src.intensities)
    for elem in src.intensities:
        py_to_cdr_body_LaserEcho(writer, elem)


def cdr_body_to_py_MultiEchoLaserScan(reader: CdrReader, dst: MultiEchoLaserScan):
    cdr_body_to_py_Header(reader, dst.header)
    dst.angle_min = reader.read_float32()
    dst.angle_max = reader.read_float32()
    dst.angle_increment = reader.read_float32()
    dst.time_increment = reader.read_float32()
    dst.scan_time = reader.read_float32()
    dst.range_min = reader.read_float32()
    dst.range_max = reader.read_float32()
    dst.ranges = []
    for _ in range(reader.read_uint32()):
        elem = LaserEcho()
        cdr_body_to_py_LaserEcho(reader, elem)
        dst.ranges.append(elem)
    dst.intensities = []
    for _ in range(reader.read_uint32()):
        elem = LaserEcho()
        cdr_body_to_py_LaserEcho(reader, elem)
        dst.intensities.append(elem)
    return dst


def py_to_cdr_MultiEchoLaserScan(src: MultiEchoLaserScan) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_MultiEchoLaserScan(writer, src)
    return writer.bytes()


def cdr_to_py_MultiEchoLaserScan(cdr_payload) -> MultiEchoLaserScan:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = MultiEchoLaserScan()
    return cdr_body_to_py_MultiEchoLaserScan(reader, dst)

File: pdu/javascript/sensor_msgs/pdu_cdr_conv_MultiEchoLaserScan.js
import { MultiEchoLaserScan } from './pdu_jstype_MultiEchoLaserScan.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';
import { Header } from '../std_msgs/pdu_jstype_Header.js';
import { PduHeaderConverter } from '../std_msgs/pdu_cdr_conv_Header.js';
import { LaserEcho } from '../sensor_msgs/pdu_jstype_LaserEcho.js';
import { PduLaserEchoConverter } from '../sensor_msgs/pdu_cdr_conv_LaserEcho.js';
import { Time } from '../builtin_interfaces/pdu_jstype_Time.js';
import { PduTimeConverter } from '../builtin_interfaces/pdu_cdr_conv_Time.js';


export class PduMultiEchoLaserScanConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { MultiEchoLaserScan } src
     */
    static to_cdr_body(writer, src) {
        PduHeaderConverter.to_cdr_body(writer, src.header);
        writer.write_float32(src.angle_min);
        writer.write_float32(src.angle_max);
        writer.write_float32(src.angle_increment);
        writer.write_float32(src.time_increment);
        writer.write_float32(src.scan_time);
        writer.write_float32(src.range_min);
        writer.write_float32(src.range_max);
        writer.write_sequence_length(src.ranges);
        for (const elem of src.ranges) {
            PduLaserEchoConverter.to_cdr_body(writer, elem);
        }
        writer.write_sequence_length(src.intensities);
        for (const elem of src.intensities) {
            PduLaserEchoConverter.to_cdr_body(writer, elem);
        }
    }

    /**
     * @param {PduCdrReader} reader
     * @param { MultiEchoLaserScan } dst
     * @returns { MultiEchoLaserScan }
     */
    static cdr_body_to_js(reader, dst) {
        PduHeaderConverter.cdr_body_to_js(reader, dst.header);
        dst.angle_min = reader.read_float32();
        dst.angle_max = reader.read_float32();
        dst.angle_increment = reader.read_float32();
        dst.time_increment = reader.read_float32();
        dst.scan_time = reader.read_float32();
        dst.range_min = reader.read_float32();
        dst.range_max = reader.read_float32();
        dst.ranges = [];
        for (let i = 0, len = reader.read_uint32(); i < len; i++) {
            dst.ranges.push(PduLaserEchoConverter.cdr_body_to_js(reader, new LaserEcho()));
        }
        dst.intensities = [];
        for (let i = 0, len = reader.read_uint32(); i < len; i++) {
            dst.intensities.push(PduLaserEchoConverter.cdr_body_to_js(reader, new LaserEcho()));
        }
        return dst;
    }

    /**
     * @param { MultiEchoLaserScan } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { MultiEchoLaserScan }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new MultiEchoLaserScan());
    }
}
